packages/netlify-cms-widget-markdown/src/MarkdownControl/plugins/BreakToDefaultBlock.js

```diff
@@ -12,6 +12,11 @@ function BreakToDefaultBlock({ defaultType }) {
         editor.delete();
         return next();
       }
+      /** If the block that receives the Enter key is a child of a list item block, we'll
+       * skip this plugin and create a new list item block underneath instead.
+       */
+      const parent = document.getParent(startBlock.key);
+      if (parent.type === 'list-item') return next();
```

Comment:
Can you help me understand when this code it triggered? I couldn't find a scenario where I reach it

Reply:
Oh, this code is a remnant from another (scraped) approach where I attempted to make list item blocks contain only text nodes, not pargraph block, like this:
```
<li data-slate-object="block" data-key="143">
  <span data-slate-object="text" data-key="144">
    <span data-slate-leaf="true" data-offset-key="144:0">
      <span data-slate-string="true">Try it out for yourself!</span>
    </span>
  </span>
</li>
```

With that schema, when you hit Enter on a list item, this condition inside the `BreakToDefaultBlock` plugin will be met:

```
if (
        selection.start.isAtEndOfNode(startBlock) &&
        startBlock.type !== "paragraph"
      ) {
        return editor.insertBlock("paragraph");
      }
```
because this plugin is loaded after the List plugin. This means instead of creating a new list item, we'll get another paragraph block on the level as the previous list item.

If you're interested, check out this [demo](https://codesandbox.io/s/slate-editor-v047-ptjv8?file=/src/components/VisualEditor.js) I made a month ago. Place cursor on the final line and click Bulleted List button. Then hit Enter.

But long story short, we no longer need this 😅.

Reply:
What do you suggest I do to this code? Remove the check for `list-item` and make a new commit, or rewrite the history?

Reply:
You can remove the code in a new commit. We squash merge PRs anyway so all the commits from this PR will show up as a single commit on the default branch.